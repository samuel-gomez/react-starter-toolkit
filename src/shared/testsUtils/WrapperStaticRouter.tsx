/* eslint-disable react/display-name */
import { ReactNode } from 'react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

const WrapperStaticRouter =
  (route = '/') =>
  ({ children }: { children: ReactNode }) =>
    <MemoryRouterProvider url={route}>{children}</MemoryRouterProvider>;

export default WrapperStaticRouter;
