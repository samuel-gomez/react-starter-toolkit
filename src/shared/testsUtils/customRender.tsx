/* eslint-disable react/display-name */
import { render, RenderOptions } from '@testing-library/react';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';

import FetchProvider from 'providers/FetchProvider';
import UserProvider from 'providers/UserProvider';
import { ReactElement, ReactNode } from 'react';
import { MOCK_API_URL } from '.';

type TMockProvider = {
  [x: string]: Record<string, unknown | number | string> | string;
};

const MockProviders =
  ({ role = '', ...testMock }: TMockProvider) =>
  ({ children }: { children: ReactNode }) => {
    const useOidcAccessTokenFn = jest.fn().mockReturnValue({ accessToken: 'accessToken' });
    const useOidcUserFn = jest.fn().mockReturnValueOnce({ oidcUser: { member_of: [`CN=${role}`], name: 'Samuel Gomez' } });
    return (
      <UserProvider useOidcUserFn={useOidcUserFn}>
        <FetchProvider
          apiUrl={MOCK_API_URL}
          fetchConfig={{
            headers: { testMock: JSON.stringify(testMock) },
          }}
          useOidcAccessTokenFn={useOidcAccessTokenFn}
        >
          <MemoryRouterProvider url="">{children}</MemoryRouterProvider>
        </FetchProvider>
      </UserProvider>
    );
  };

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>, testMock: TMockProvider = { role: '' }) =>
  render(ui, { wrapper: MockProviders(testMock), ...options });

// re-export everything
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// override render method
export { customRender as render };
