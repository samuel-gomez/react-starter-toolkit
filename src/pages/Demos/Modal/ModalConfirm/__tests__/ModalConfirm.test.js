import { render } from '@testing-library/react';
import { emptyFunction } from 'shared/testsUtils';
import ModalConfirm from '../ModalConfirm';

const defaultProps = {
  isOpen: false,
  onCancel: emptyFunction,
};

describe('<ModalConfirm/>', () => {
  it('Should render ModalConfirm', () => {
    const { baseElement } = render(<ModalConfirm {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('Should render ModalConfirm opened', () => {
    const { baseElement } = render(<ModalConfirm {...defaultProps} isOpen />);
    expect(baseElement).toMatchSnapshot();
  });
});
