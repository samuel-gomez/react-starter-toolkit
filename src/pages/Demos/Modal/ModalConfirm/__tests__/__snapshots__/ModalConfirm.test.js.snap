// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalConfirm/> Should render ModalConfirm 1`] = `
<body>
  <div />
  <div
    class="ReactModalPortal"
  />
</body>
`;

exports[`<ModalConfirm/> Should render ModalConfirm opened 1`] = `
<body
  aria-hidden="true"
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(255, 255, 255, 0.75);"
    >
      <div
        aria-label=""
        aria-modal="true"
        class="ReactModal__Content af-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="af-modal__dialog"
        >
          <div
            class="af-modal__content"
          >
            <div
              class="af-modal__header"
            >
              <h4
                class="af-modal__header-title"
              >
                Validation des informations générales
              </h4>
              <button
                aria-label="Close"
                class="af-modal__header-close-btn"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="glyphicon glyphicon-close"
                />
              </button>
            </div>
            <div
              class="af-modal__body"
            >
              <div
                class="af-alert af-alert--info"
                role="alert"
              >
                <div
                  class="af-alert__title"
                >
                  <div
                    class="af-alert__title-icon"
                  >
                    <i
                      class="glyphicon glyphicon-info-sign"
                    />
                  </div>
                  <div
                    class="af-alert__title-text"
                  >
                    Vous allez créer un nouvel élément.
                  </div>
                </div>
              </div>
              <p>
                Confirmez-vous vouloir valider avec les informations suivantes ?
              </p>
            </div>
            <div
              class="af-modal__footer"
            >
              <button
                class="btn af-btn af-btn--reverse"
              >
                <span
                  class="af-btn__text"
                >
                  Annuler
                </span>
              </button>
              <button
                class="btn af-btn af-btn--hasiconRight"
                type="button"
              >
                <span
                  class="af-btn__text"
                >
                  Valider
                </span>
                <i
                  class="glyphicon glyphicon-arrowthin-right"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
